lib/internal/Magento/Framework/HTTP/Client/Curl.php

```diff
@@ -338,17 +338,18 @@ public function getStatus()
      * @param string $method
      * @param string $uri
      * @param array $params
+     * @param bool $json
      * @return void
      * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      * @SuppressWarnings(PHPMD.NPathComplexity)
      */
-    protected function makeRequest($method, $uri, $params = [])
+    protected function makeRequest($method, $uri, $params = [], $json = false)
```

Comment:
Thank you for the fixes. Unfortunately, by this fix you modified public interface of the class and added knowledge about JSON for Curl client. On the other hand, how to post XML request for instance? We don't want to add $isXML flag to the method again. So we need to have something more general - string format.

Let me suggest another approach:
```
 * @param array|string $params
protected function makeReques (...) {
 
   ...
   $this->curlOption(CURLOPT_POSTFIELDS, is_array($params) ? http_build_query($params) : $params);
   ...
}
```
We are extending our contract by adding new supported type *string* wihtout changes in public method parameters. 

Please, do not forget to align makeRequest and post method signatures. 

Thank you for reporting and fixing this issue. Good catch! +100500 to your karma! ;)